src/client/main.cpp

```diff
@@ -133,13 +170,118 @@ int ioctl(int fd, unsigned long request, char *ptr) {
   return func_ioctl(fd, request, ptr);
 }
 
+static const auto touchArgs = QByteArray("rotate=180:invertx");
+
+bool _Z7qputenvPKcRK10QByteArray(const char *name, const QByteArray &val) {
+  static auto orig_fn = (bool (*)(const char *, const QByteArray &))dlsym(
+      RTLD_NEXT, "_Z7qputenvPKcRK10QByteArray");
+
+  if (strcmp(name, "QT_QPA_EVDEV_TOUCHSCREEN_PARAMETERS") == 0) {
+    return orig_fn(name, touchArgs);
+  }
+
+  return orig_fn(name, val);
+}
+
+void (*old_update)(void *, int, int, int, int, int, int) = 0;
+int (*old_create_threads)(const char *, void *) = 0;
+int (*old_wait)(void) = 0;
+int (*old_shutdown)(void) = 0;
+
+void new_update(void *instance, int x1, int y1, int x2, int y2, int waveform,
+                int flags) {
+#ifdef DEBUG
+  std::cerr << "UPDATE HOOK CALLED" << std::endl;
+  std::cerr << "x " << x1 << " " << x2 << std::endl;
+  std::cerr << "y " << y1 << " " << y2 << std::endl;
+  std::cerr << "wav " << waveform << " flags " << flags << std::endl;
+#endif
+
+  swtfb::xochitl_data data;
+  data.x1 = x1;
+  data.x2 = x2;
+  data.y1 = y1;
+  data.y2 = y2;
+  data.waveform = waveform;
+  data.flags = flags;
+  MSGQ.send(data);
+}
+
+int new_create_threads(const char *path, void *buf) {
+  std::cerr << "create threads called" << std::endl;
+  return 0;
+}
+
+int new_wait(void) {
+  std::cerr << "wait clear func called" << std::endl;
+  return 0;
+}
+
+int new_shutdown(void) {
+  std::cerr << "shutdown called" << std::endl;
+  return 0;
+}
+
+GumInterceptor *interceptor;
+
 int __libc_start_main(int (*_main)(int, char **, char **), int argc,
                       char **argv, int (*init)(int, char **, char **),
                       void (*fini)(void), void (*rtld_fini)(void),
                       void *stack_end) {
 
-  if (string(argv[0]).find("xochitl") != string::npos) {
+  if (std::string(argv[0]).find("xochitl") != std::string::npos) {
```

Comment:
technically, it should be that xochitl appears in the first argument of argv[0], this will catch a proc like `grep xochitl`

Reply:
It uses `argv[0]` right, which would be the program name.
in `grep xochitl`, argv[0] would be `grep` and argv[1] would be `xochitl`

Reply:
oops, good point! i should have tested / checked before commenting